### PR TITLE
replace previous user listing with jquery datatables

### PR DESCRIPTION
**Pull request by debugteam**
I strongly request a review on translations.

### CODE REVIEW

**Comment by LazerTiberius** [comment]
File FOM/UserBundle/Ressources/public/sass/user_control.scss is missing

**Comment by AxxL** [comment]
@debugteam 

Error opening users page:

> DataTables warning: table id=listFilterGroups - Ajax error. For more information about this error, please see http://datatables.net/tn/

**Comment by debugteam** [comment]
I've just made a fresh installation with the following steps on Ubuntu 16.4 LTS (Xenial Xerus):

1. created a folder in /var/www/vhosts/fomext/web
2. cd /var/www/vhosts/fomext/web
3. git clone git@github.com:mapbender/mapbender-starter.git ./
4. set up a new vhost with document root to /var/www/vhosts/fomext/web/application/web
5. restarted webserver
6. cd application
7. vi composer.json 
8. changed fom requirement to: "mapbender/fom": "dev-feature/manager-user-datatables",
9. esc + :wq (to save the file *g*)
10. composer update -vvv  ( -vvv just to make sure everything installs correctly)
11. cp app/config/parameters.yml.dist app/config/parameters.yml
12. app/console doctrine:database:create
13. app/console doctrine:schema:create
14. app/console assets:install web
15. app/console fom:user:resetroot
16. app/console doctrine:fixtures:load --fixtures=./mapbender/src/Mapbender/CoreBundle/DataFixtures/ORM/Epsg/ --append
17. app/console doctrine:fixtures:load --fixtures=./mapbender/src/Mapbender/CoreBundle/DataFixtures/ORM/Application/ --append

$ git branch
* release/3.0.6

log in on http://fomext.dev (due to having a servername in my vhost config and dns entry this links to the documentroot)

click on "benutzer" ... et voilà

File: src/FOM/UserBundle/Resources/public/css/userlist.datatables.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.dataTables_filter label input {
```
Comment (eSlider) [inline src/FOM/UserBundle/Resources/public/css/userlist.datatables.css]: * This is a global declaration for DataTables and not only for users, but file is named as `userlist.datatables.css`
* A lot of `!important` isn't good.
* `.tdsmall`, `.dataTable` selector definitions as global defintion isn't good
* Please change it to SCSS with variable definitions

File: src/FOM/UserBundle/Resources/translations/messages.de.xlf [hunk 2]
```diff
@@ -594,6 +594,54 @@
         <source>User credentials have expired.</source>
         <target>Anmeldedaten sind abgelaufen.</target>
       </trans-unit>
+      <trans-unit id="fom.user.user.index.datatable.emptytable">
```
Comment (eSlider) [inline src/FOM/UserBundle/Resources/translations/messages.de.xlf]: * Translation definition  `fom.user.user.index.datatable.emptytable` should be `mb.manager.admin.user.no-results`
* Translation should be an YAML file, XLF isn't wellcome
* `datatable` as key name namespace isn't necessary. What if we change DataTables to another plugin? 
* `user.user` - really?
*  FOM Bundle will be splitted in several modules, so please no `fom` any more.

Take a look at current administration translations defined as an YAML:
https://github.com/mapbender/mapbender/blob/release/3.0.6/src/Mapbender/ManagerBundle/Resources/translations/messages.de.yml#L30

Please rethink and change translations.

File: src/FOM/UserBundle/Resources/views/User/index.html.twig [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 {% block css %}
   {{parent()}}
+  <link rel="stylesheet" href="/components/datatables/datatables-built.css" />
```
Comment (eSlider) [inline src/FOM/UserBundle/Resources/views/User/index.html.twig]: `<link rel="stylesheet" href="{{ asset('components/datatables/datatables-built.css') }}" />`

File: src/FOM/UserBundle/Controller/UserController.php [hunk 4]
```diff
@@ -30,21 +33,71 @@ class UserController extends Controller
...
+     */
+    public function searchAction()
+    {
+        $securityContext = $this->get('security.context');
+        $oid = new ObjectIdentity('class', 'FOM\UserBundle\Entity\User');
+        $request = Request::createFromGlobals();
+        $start = $request->query->get('start');
+        $length = $request->query->get('length');
+        $search = $request->query->get('search');
+        $dql = "SELECT r FROM FOMUserBundle:User r LEFT JOIN r.groups g WHERE
```
Comment (eSlider) [inline src/FOM/UserBundle/Controller/UserController.php]: Please use/improve `FOMIdentitiesProvider` for user managing/handling. This is a global user service: ` $this->container->get('fom.identities.provider')`
Path: `application/fom/src/FOM/UserBundle/Component/FOMIdentitiesProvider.php`

File: src/FOM/UserBundle/Resources/translations/messages.de.xlf (hunk 2 above)
Comment (debugteam) [inline src/FOM/UserBundle/Resources/translations/messages.de.xlf]: I just added some translations to existing Translation in FOM. I suggest you open an enhancement issue for that (like change xlf to yaml). 

fom.user.user... yep that's what i found there :)

File: src/FOM/UserBundle/Controller/UserController.php (hunk 4 above)
Comment (debugteam) [inline src/FOM/UserBundle/Controller/UserController.php]: I suggest you open an issue for that (enhancement - replace ObjectIdentity with FOMIdentitiesProvider).
My intention wasn't a complete refactoring of the code...

File: src/FOM/UserBundle/Resources/public/css/userlist.datatables.css (hunk 1 above)
Comment (debugteam) [inline src/FOM/UserBundle/Resources/public/css/userlist.datatables.css]: The userlist.datatables.css is only included when you open the userlist and should totally override the global datatables CSS definition thus !important is a possible way to go. 

yep, .tdsmall ain't a good name for a class omg

Please open an issue for the SCSS as well.

File: src/FOM/UserBundle/Resources/views/User/index.html.twig (hunk 3 above)
Comment (debugteam) [inline src/FOM/UserBundle/Resources/views/User/index.html.twig]: Yeah, sorry. Will change that. My bad.

File: src/FOM/UserBundle/Resources/translations/messages.de.xlf (hunk 2 above)
Comment (eSlider) [inline src/FOM/UserBundle/Resources/translations/messages.de.xlf]: @debugteam, please use, at least, right naming of translation

File: src/FOM/UserBundle/Resources/public/css/userlist.datatables.css (hunk 1 above)
Comment (eSlider) [inline src/FOM/UserBundle/Resources/public/css/userlist.datatables.css]: @debugteam, 
1. I will not open an issue for that
2. Stop to use `!important` declaration
3. No global selectors like `.tdsmall`

File: src/FOM/UserBundle/Controller/UserController.php (hunk 4 above)
Comment (eSlider) [inline src/FOM/UserBundle/Controller/UserController.php]: It's not refactoring, but improvement.

**Comment by LazerTiberius** [comment]
I will do some work on this here https://github.com/mapbender/fom/tree/feature/userManagerwithDataTables for 3.0.5 and later based on this for 3.0.6

**Pull request closed without merging**